[Autotune Cache] Fix the bug of using the wrong key for recording artifacts in CacheArtifactManager

dongji-gao opened the pull request:
Summary: Replace the key (path) from `<hash>.best_config` to `<parent_dir>/<hash>.best_config` to ensure that Autotune artifacts in MegaCache are loaded to the correct location locally.

Test Plan: NA

Differential Revision: D74052400




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

oulgen commented on torch/_inductor/runtime/autotune_cache.py:
```diff
@@ -238,8 +238,10 @@ def save(
             cache, key = local_cache
             cache.put(key, data)
             AutotuneCacheBundler.put(key, data)
+            key_parts = os.path.normpath(key).split(os.sep)
```
I dont think you need normpath, it is already normalized

dongji-gao committed:
[Autotune Cache] Fix the bug of using the wrong key for recording artifacts in CacheArtifactManager (#152678)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/152678

Replace the key (path) from `<hash>.best_config` to `<parent_dir>/<hash>.best_config` to ensure that Autotune artifacts in MegaCache are loaded to the correct location locally.

Test Plan: NA

Reviewed By: oulgen

Differential Revision: D74052400

dongji-gao commented:
@pytorchbot merge
